test/e2e/topology_updater.go

```diff
@@ -79,27 +78,52 @@ var _ = ginkgo.Describe("[TopologyUpdater][InfraConsuming] Node topology updater
 	})
 
 	ginkgo.Context("with cluster configured", func() {
-		ginkgo.It("it should not account for any cpus if a container doesn't request exclusive cpus", func() {
+		ginkgo.It("it should not account for any cpus if a container doesn't request exclusive cpus (best effort QOS)", func() {
 			ginkgo.By("getting the initial topology information")
 			initialNodeTopo := getNodeTopology(topologyClient, topologyUpdaterNode.Name, namespace)
-			ginkgo.By("creating a pod consuming the shared pool")
-			sleeperPod := &v1.Pod{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "sleeper-be-pod",
-				},
-				Spec: v1.PodSpec{
-					RestartPolicy: v1.RestartPolicyNever,
-					Containers: []v1.Container{
-						v1.Container{
-							Name:  "sleeper-be-cnt",
-							Image: utils.CentosImage,
-							// 1 hour (or >= 1h in general) is "forever" for our purposes
-							Command: []string{"/bin/sleep", "1h"},
-						},
-					},
-				},
+			ginkgo.By("creating a pod consuming resources from the shared, non-exclusive CPU pool (best-effort QoS)")
+			sleeperPod := utils.MakeBestEffortSleeperPod()
+
+			podMap := make(map[string]*v1.Pod)
+			pod := f.PodClient().CreateSync(sleeperPod)
+			podMap[pod.Name] = pod
+			defer utils.DeletePodsAsync(f, podMap)
+
+			cooldown := 30 * time.Second
+			ginkgo.By(fmt.Sprintf("getting the updated topology - sleeping for %v", cooldown))
+			// the object, hance the resource version must NOT change, so we can only sleep
+			time.Sleep(cooldown)
+			ginkgo.By("checking the changes in the updated topology - expecting none")
+			finalNodeTopo := getNodeTopology(topologyClient, topologyUpdaterNode.Name, namespace)
+
+			initialAllocRes := allocatableResourceListFromNodeResourceTopology(initialNodeTopo)
+			finalAllocRes := allocatableResourceListFromNodeResourceTopology(finalNodeTopo)
+			if len(initialAllocRes) == 0 || len(finalAllocRes) == 0 {
+				ginkgo.Fail(fmt.Sprintf("failed to find allocatable resources from node topology initial=%v final=%v", initialAllocRes, finalAllocRes))
+			}
+			zoneName, resName, cmp, ok := cmpAllocatableResources(initialAllocRes, finalAllocRes)
+			framework.Logf("zone=%q resource=%q cmp=%v ok=%v", zoneName, resName, cmp, ok)
+			if !ok {
+				ginkgo.Fail(fmt.Sprintf("failed to compare allocatable resources from node topology initial=%v final=%v", initialAllocRes, finalAllocRes))
 			}
 
+			// This is actually a workaround.
```

Comment:
This is definitely a TODO for us but appreciate you providing detailed explanation here.

Reply:
ack, will mark it as such